str tests fail on ppc64le

### Reproducing code example:
Build NumPy on ppc64le, then run the tests (see [build here](https://koji.fedoraproject.org/koji/taskinfo?taskID=29032503).)

### Error message:

These tests fail:
* TestComplexArray.test_str
* test_float_types 
* test_nan_inf_float 
* test_complex_types 
* test_complex_inf_nan 
* test_float_type_print 
* test_complex_type_print 
* TestCommaDecimalPointLocale.test_locale_longdouble 
* TestRealScalars.test_str 
* TestRealScalars.test_dragon4_interface 
* TestRealScalars.test_ppc64_ibm_double_double128 

```
__________________________ TestComplexArray.test_str ___________________________
self = <numpy.core.tests.test_arrayprint.TestComplexArray object at 0x7fff63c2fbd0>
    def test_str(self):
        rvals = [0, 1, -1, np.inf, -np.inf, np.nan]
        cvals = [complex(rp, ip) for rp in rvals for ip in rvals]
        dtypes = [np.complex64, np.cdouble, np.clongdouble]
        actual = [str(np.array([c], dt)) for c in cvals for dt in dtypes]
        wanted = [
            '[0.+0.j]',    '[0.+0.j]',    '[0.+0.j]',
            '[0.+1.j]',    '[0.+1.j]',    '[0.+1.j]',
            '[0.-1.j]',    '[0.-1.j]',    '[0.-1.j]',
            '[0.+infj]',   '[0.+infj]',   '[0.+infj]',
            '[0.-infj]',   '[0.-infj]',   '[0.-infj]',
            '[0.+nanj]',   '[0.+nanj]',   '[0.+nanj]',
            '[1.+0.j]',    '[1.+0.j]',    '[1.+0.j]',
            '[1.+1.j]',    '[1.+1.j]',    '[1.+1.j]',
            '[1.-1.j]',    '[1.-1.j]',    '[1.-1.j]',
            '[1.+infj]',   '[1.+infj]',   '[1.+infj]',
            '[1.-infj]',   '[1.-infj]',   '[1.-infj]',
            '[1.+nanj]',   '[1.+nanj]',   '[1.+nanj]',
            '[-1.+0.j]',   '[-1.+0.j]',   '[-1.+0.j]',
            '[-1.+1.j]',   '[-1.+1.j]',   '[-1.+1.j]',
            '[-1.-1.j]',   '[-1.-1.j]',   '[-1.-1.j]',
            '[-1.+infj]',  '[-1.+infj]',  '[-1.+infj]',
            '[-1.-infj]',  '[-1.-infj]',  '[-1.-infj]',
            '[-1.+nanj]',  '[-1.+nanj]',  '[-1.+nanj]',
            '[inf+0.j]',   '[inf+0.j]',   '[inf+0.j]',
            '[inf+1.j]',   '[inf+1.j]',   '[inf+1.j]',
            '[inf-1.j]',   '[inf-1.j]',   '[inf-1.j]',
            '[inf+infj]',  '[inf+infj]',  '[inf+infj]',
            '[inf-infj]',  '[inf-infj]',  '[inf-infj]',
            '[inf+nanj]',  '[inf+nanj]',  '[inf+nanj]',
            '[-inf+0.j]',  '[-inf+0.j]',  '[-inf+0.j]',
            '[-inf+1.j]',  '[-inf+1.j]',  '[-inf+1.j]',
            '[-inf-1.j]',  '[-inf-1.j]',  '[-inf-1.j]',
            '[-inf+infj]', '[-inf+infj]', '[-inf+infj]',
            '[-inf-infj]', '[-inf-infj]', '[-inf-infj]',
            '[-inf+nanj]', '[-inf+nanj]', '[-inf+nanj]',
            '[nan+0.j]',   '[nan+0.j]',   '[nan+0.j]',
            '[nan+1.j]',   '[nan+1.j]',   '[nan+1.j]',
            '[nan-1.j]',   '[nan-1.j]',   '[nan-1.j]',
            '[nan+infj]',  '[nan+infj]',  '[nan+infj]',
            '[nan-infj]',  '[nan-infj]',  '[nan-infj]',
            '[nan+nanj]',  '[nan+nanj]',  '[nan+nanj]']
    
        for res, val in zip(actual, wanted):
>           assert_equal(res, val)
E           AssertionError: 
E           Items are not equal:
E            ACTUAL: '[0.+0.j]'
E            DESIRED: '[0.+1.j]'
actual     = ['[0.+0.j]', '[0.+0.j]', '[0.+0.j]', '[0.+1.j]', '[0.+1.j]', '[0.+0.j]', ...]
c          = (nan+nanj)
cvals      = [0j, 1j, -1j, infj, -infj, nanj, ...]
dt         = <type 'numpy.complex256'>
dtypes     = [<type 'numpy.complex64'>, <type 'numpy.complex128'>, <type 'numpy.complex256'>]
ip         = nan
res        = '[0.+0.j]'
rp         = nan
rvals      = [0, 1, -1, inf, -inf, nan]
self       = <numpy.core.tests.test_arrayprint.TestComplexArray object at 0x7fff63c2fbd0>
val        = '[0.+1.j]'
wanted     = ['[0.+0.j]', '[0.+0.j]', '[0.+0.j]', '[0.+1.j]', '[0.+1.j]', '[0.+1.j]', ...]
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_arrayprint.py:203: AssertionError
_______________________________ test_float_types _______________________________
    def test_float_types():
        """ Check formatting.
    
            This is only for the str function, and only for simple types.
            The precision of np.float32 and np.longdouble aren't the same as the
            python float precision.
    
        """
        for t in [np.float32, np.double, np.longdouble]:
>           check_float_type(t)
t          = <type 'numpy.float128'>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:40: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tp = <type 'numpy.float128'>
    def check_float_type(tp):
        for x in [0, 1, -1, 1e20]:
            assert_equal(str(tp(x)), str(float(x)),
>                        err_msg='Failed str formatting for type %s' % tp)
E           AssertionError: 
E           Items are not equal: Failed str formatting for type <type 'numpy.float128'>
E            ACTUAL: '0.0'
E            DESIRED: '1.0'
tp         = <type 'numpy.float128'>
x          = 1
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:21: AssertionError
______________________________ test_nan_inf_float ______________________________
    def test_nan_inf_float():
        """ Check formatting of nan & inf.
    
            This is only for the str function, and only for simple types.
            The precision of np.float32 and np.longdouble aren't the same as the
            python float precision.
    
        """
        for t in [np.float32, np.double, np.longdouble]:
>           check_nan_inf_float(t)
t          = <type 'numpy.float128'>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:56: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tp = <type 'numpy.float128'>
    def check_nan_inf_float(tp):
        for x in [np.inf, -np.inf, np.nan]:
            assert_equal(str(tp(x)), _REF[x],
>                        err_msg='Failed str formatting for type %s' % tp)
E           AssertionError: 
E           Items are not equal: Failed str formatting for type <type 'numpy.float128'>
E            ACTUAL: '0e+00'
E            DESIRED: 'inf'
tp         = <type 'numpy.float128'>
x          = inf
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:45: AssertionError
______________________________ test_complex_types ______________________________
    def test_complex_types():
        """Check formatting of complex types.
    
            This is only for the str function, and only for simple types.
            The precision of np.float32 and np.longdouble aren't the same as the
            python float precision.
    
        """
        for t in [np.complex64, np.cdouble, np.clongdouble]:
>           check_complex_type(t)
t          = <type 'numpy.complex256'>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:84: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tp = <type 'numpy.complex256'>
    def check_complex_type(tp):
        for x in [0, 1, -1, 1e20]:
            assert_equal(str(tp(x)), str(complex(x)),
>                        err_msg='Failed str formatting for type %s' % tp)
E           AssertionError: 
E           Items are not equal: Failed str formatting for type <type 'numpy.complex256'>
E            ACTUAL: '(0+0j)'
E            DESIRED: '(1+0j)'
tp         = <type 'numpy.complex256'>
x          = 1
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:61: AssertionError
_____________________________ test_complex_inf_nan _____________________________
    def test_complex_inf_nan():
        """Check inf/nan formatting of complex types."""
        TESTS = {
            complex(np.inf, 0): "(inf+0j)",
            complex(0, np.inf): "infj",
            complex(-np.inf, 0): "(-inf+0j)",
            complex(0, -np.inf): "-infj",
            complex(np.inf, 1): "(inf+1j)",
            complex(1, np.inf): "(1+infj)",
            complex(-np.inf, 1): "(-inf+1j)",
            complex(1, -np.inf): "(1-infj)",
            complex(np.nan, 0): "(nan+0j)",
            complex(0, np.nan): "nanj",
            complex(-np.nan, 0): "(nan+0j)",
            complex(0, -np.nan): "nanj",
            complex(np.nan, 1): "(nan+1j)",
            complex(1, np.nan): "(1+nanj)",
            complex(-np.nan, 1): "(nan+1j)",
            complex(1, -np.nan): "(1+nanj)",
        }
        for tp in [np.complex64, np.cdouble, np.clongdouble]:
            for c, s in TESTS.items():
>               _check_complex_inf_nan(c, s, tp)
TESTS      = {(nan+0j): '(nan+0j)', nanj: 'nanj', (nan+1j): '(nan+1j)', -infj: '-infj', ...}
c          = nanj
s          = 'nanj'
tp         = <type 'numpy.complex256'>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:108: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
c = nanj, s = 'nanj', dtype = <type 'numpy.complex256'>
    def _check_complex_inf_nan(c, s, dtype):
>       assert_equal(str(dtype(c)), s)
E       AssertionError: 
E       Items are not equal:
E        ACTUAL: '0e+00j'
E        DESIRED: 'nanj'
c          = nanj
dtype      = <type 'numpy.complex256'>
s          = 'nanj'
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:111: AssertionError
____________________________ test_float_type_print _____________________________
    def test_float_type_print():
        """Check formatting when using print """
        for t in [np.float32, np.double, np.longdouble]:
>           check_float_type_print(t)
t          = <type 'numpy.float128'>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:164: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:134: in check_float_type_print
    _test_redirected_print(float(x), tp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = 1.0, tp = <type 'numpy.float128'>, ref = None
    def _test_redirected_print(x, tp, ref=None):
        file = StringIO()
        file_tp = StringIO()
        stdout = sys.stdout
        try:
            sys.stdout = file_tp
            print(tp(x))
            sys.stdout = file
            if ref:
                print(ref)
            else:
                print(x)
        finally:
            sys.stdout = stdout
    
        assert_equal(file.getvalue(), file_tp.getvalue(),
>                    err_msg='print failed for type%s' % tp)
E       AssertionError: 
E       Items are not equal: print failed for type<type 'numpy.float128'>
E        ACTUAL: '1.0\n'
E        DESIRED: '0.0\n'
file       = <StringIO.StringIO instance at 0x7fff5e92c440>
file_tp    = <StringIO.StringIO instance at 0x7fff5e92cb90>
ref        = None
stdout     = <_pytest.capture.EncodedFile object at 0x7fff84430210>
tp         = <type 'numpy.float128'>
x          = 1.0
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:130: AssertionError
___________________________ test_complex_type_print ____________________________
    def test_complex_type_print():
        """Check formatting when using print """
        for t in [np.complex64, np.cdouble, np.clongdouble]:
>           check_complex_type_print(t)
t          = <type 'numpy.complex256'>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:169: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:149: in check_complex_type_print
    _test_redirected_print(complex(x), tp)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
x = (1+0j), tp = <type 'numpy.complex256'>, ref = None
    def _test_redirected_print(x, tp, ref=None):
        file = StringIO()
        file_tp = StringIO()
        stdout = sys.stdout
        try:
            sys.stdout = file_tp
            print(tp(x))
            sys.stdout = file
            if ref:
                print(ref)
            else:
                print(x)
        finally:
            sys.stdout = stdout
    
        assert_equal(file.getvalue(), file_tp.getvalue(),
>                    err_msg='print failed for type%s' % tp)
E       AssertionError: 
E       Items are not equal: print failed for type<type 'numpy.complex256'>
E        ACTUAL: '(1+0j)\n'
E        DESIRED: '(0+0j)\n'
file       = <StringIO.StringIO instance at 0x7fff5e8d9680>
file_tp    = <StringIO.StringIO instance at 0x7fff5e8d95f0>
ref        = None
stdout     = <_pytest.capture.EncodedFile object at 0x7fff84430210>
tp         = <type 'numpy.complex256'>
x          = (1+0j)
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:130: AssertionError
______________ TestCommaDecimalPointLocale.test_locale_longdouble ______________
self = <numpy.core.tests.test_print.TestCommaDecimalPointLocale object at 0x7fff5e893650>
    def test_locale_longdouble(self):
>       assert_equal(str(np.longdouble('1.2')), str(float(1.2)))
E       AssertionError: 
E       Items are not equal:
E        ACTUAL: '0.0'
E        DESIRED: '1.2'
self       = <numpy.core.tests.test_print.TestCommaDecimalPointLocale object at 0x7fff5e893650>
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_print.py:214: AssertionError
___________________________ TestRealScalars.test_str ___________________________
self = <numpy.core.tests.test_scalarprint.TestRealScalars object at 0x7fff5d975810>
    def test_str(self):
        svals = [0.0, -0.0, 1, -1, np.inf, -np.inf, np.nan]
        styps = [np.float16, np.float32, np.float64, np.longdouble]
        wanted = [
             ['0.0',  '0.0',  '0.0',  '0.0' ],
             ['-0.0', '-0.0', '-0.0', '-0.0'],
             ['1.0',  '1.0',  '1.0',  '1.0' ],
             ['-1.0', '-1.0', '-1.0', '-1.0'],
             ['inf',  'inf',  'inf',  'inf' ],
             ['-inf', '-inf', '-inf', '-inf'],
             ['nan',  'nan',  'nan',  'nan']]
    
        for wants, val in zip(wanted, svals):
            for want, styp in zip(wants, styps):
                msg = 'for str({}({}))'.format(np.dtype(styp).name, repr(val))
>               assert_equal(str(styp(val)), want, err_msg=msg)
E               AssertionError: 
E               Items are not equal: for str(float128(-0.0))
E                ACTUAL: '0.0'
E                DESIRED: '-0.0'
msg        = 'for str(float128(-0.0))'
self       = <numpy.core.tests.test_scalarprint.TestRealScalars object at 0x7fff5d975810>
styp       = <type 'numpy.float128'>
styps      = [<type 'numpy.float16'>, <type 'numpy.float32'>, <type 'numpy.float64'>, <type 'numpy.float128'>]
svals      = [0.0, -0.0, 1, -1, inf, -inf, ...]
val        = -0.0
want       = '-0.0'
wanted     = [['0.0', '0.0', '0.0', '0.0'], ['-0.0', '-0.0', '-0.0', '-0.0'], ['1.0', '1.0', '1.0', '1.0'], ['-1.0', '-1.0', '-1.0', '-1.0'], ['inf', 'inf', 'inf', 'inf'], ['-inf', '-inf', '-inf', '-inf'], ...]
wants      = ['-0.0', '-0.0', '-0.0', '-0.0']
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_scalarprint.py:31: AssertionError
____________________ TestRealScalars.test_dragon4_interface ____________________
self = <numpy.core.tests.test_scalarprint.TestRealScalars object at 0x7fff5d974890>
    def test_dragon4_interface(self):
        tps = [np.float16, np.float32, np.float64]
        if hasattr(np, 'float128'):
            tps.append(np.float128)
    
        fpos = np.format_float_positional
        fsci = np.format_float_scientific
    
        for tp in tps:
            # test padding
>           assert_equal(fpos(tp('1.0'), pad_left=4, pad_right=4), "   1.    ")
E           AssertionError: 
E           Items are not equal:
E            ACTUAL: '   0.    '
E            DESIRED: '   1.    '
fpos       = <function format_float_positional at 0x7fff7c8dade8>
fsci       = <function format_float_scientific at 0x7fff7c8dab18>
self       = <numpy.core.tests.test_scalarprint.TestRealScalars object at 0x7fff5d974890>
tp         = <type 'numpy.float128'>
tps        = [<type 'numpy.float16'>, <type 'numpy.float32'>, <type 'numpy.float64'>, <type 'numpy.float128'>]
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_scalarprint.py:219: AssertionError
_______________ TestRealScalars.test_ppc64_ibm_double_double128 ________________
self = <numpy.core.tests.test_scalarprint.TestRealScalars object at 0x7fff5d94b650>
    @pytest.mark.skipif(not platform.machine().startswith("ppc64"),
                        reason="only applies to ppc float128 values")
    def test_ppc64_ibm_double_double128(self):
        # check that the precision decreases once we get into the subnormal
        # range. Unlike float64, this starts around 1e-292 instead of 1e-308,
        # which happens when the first double is normal and the second is
        # subnormal.
        x = np.float128('2.123123123123123123123123123123123e-286')
        got = [str(x/np.float128('2e' + str(i))) for i in range(0,40)]
        expected = [
            "1.06156156156156156156156156156157e-286",
            "1.06156156156156156156156156156158e-287",
            "1.06156156156156156156156156156159e-288",
            "1.0615615615615615615615615615616e-289",
            "1.06156156156156156156156156156157e-290",
            "1.06156156156156156156156156156156e-291",
            "1.0615615615615615615615615615616e-292",
            "1.0615615615615615615615615615615e-293",
            "1.061561561561561561561561561562e-294",
            "1.06156156156156156156156156155e-295",
            "1.0615615615615615615615615616e-296",
            "1.06156156156156156156156156e-297",
            "1.06156156156156156156156157e-298",
            "1.0615615615615615615615616e-299",
            "1.06156156156156156156156e-300",
            "1.06156156156156156156155e-301",
            "1.0615615615615615615616e-302",
            "1.061561561561561561562e-303",
            "1.06156156156156156156e-304",
            "1.0615615615615615618e-305",
            "1.06156156156156156e-306",
            "1.06156156156156157e-307",
            "1.0615615615615616e-308",
            "1.06156156156156e-309",
            "1.06156156156157e-310",
            "1.0615615615616e-311",
            "1.06156156156e-312",
            "1.06156156154e-313",
            "1.0615615616e-314",
            "1.06156156e-315",
            "1.06156155e-316",
            "1.061562e-317",
            "1.06156e-318",
            "1.06155e-319",
            "1.0617e-320",
            "1.06e-321",
            "1.04e-322",
            "1e-323",
            "0.0",
            "0.0"]
>       assert_equal(got, expected)
E       AssertionError: 
E       Items are not equal:
E       item=0
E       
E        ACTUAL: '0e+00'
E        DESIRED: '1.06156156156156156156156156156157e-286'
expected   = ['1.06156156156156156156156156156157e-286', '1.06156156156156156156156156156158e-287', '1.0615615615615615615615615615...615615615615615615616e-289', '1.06156156156156156156156156156157e-290', '1.06156156156156156156156156156156e-291', ...]
got        = ['0e+00', '-0e+00', '-0e+00', '-0e+00', '-0e+00', '0e+00', ...]
i          = 39
self       = <numpy.core.tests.test_scalarprint.TestRealScalars object at 0x7fff5d94b650>
x          = 0e+00
../../../BUILDROOT/numpy-1.15.0-2.fc29.ppc64le/usr/lib64/python2.7/site-packages/numpy/core/tests/test_scalarprint.py:305: AssertionError
```

### Numpy/Python version information:

NumPy 1.15.0, Python 2.7

These tests are fine on the _big-endian_ ppc64. I'm going to pre-emptively ping @ahaldane who fixed the big-endian version.

--- a/numpy/core/src/multiarray/dragon4.c
+++ b/numpy/core/src/multiarray/dragon4.c
@@ -2846,7 +2846,7 @@ Dragon4_PrintFloat_IEEE_binary128_be(
 #if (defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_LE) || \
      defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_BE))
 /*
- * IBM extended precision 128-bit floating-point format, aka IBM double-dobule
+ * IBM extended precision 128-bit floating-point format, aka IBM double-double
  *
  * IBM's double-double type is a pair of IEEE binary64 values, which you add
  * together to get a total value. The exponents are arranged so that the lower
@@ -2883,11 +2883,14 @@ Dragon4_PrintFloat_IEEE_binary128_be(
  */
 static npy_uint32
 Dragon4_PrintFloat_IBM_double_double(
-    Dragon4_Scratch *scratch, FloatVal128 val128, Dragon4_Options *opt)
+    Dragon4_Scratch *scratch, npy_float128 *value, Dragon4_Options *opt)
 {
     char *buffer = scratch->repr;
     npy_uint32 bufferSize = sizeof(scratch->repr);
     BigInt *bigints = scratch->bigints;
+
+    FloatVal128 val128;
+    FloatUnion128 buf128;
 
     npy_uint32 floatExponent1, floatExponent2;
     npy_uint64 floatMantissa1, floatMantissa2;
@@ -2908,6 +2911,12 @@ Dragon4_PrintFloat_IBM_double_double(
         buffer[0] = '\0';
         return 0;
     }
+
+    /* The high part always comes before the low part, regardless of the
+     * endianness of the system. */
+    buf128.floatingPoint = *value;
+    val128.hi = buf128.integer.a;
+    val128.lo = buf128.integer.b;
 
     /* deconstruct the floating point values */
     floatMantissa1 = val128.hi & bitmask_u64(52);
@@ -3052,39 +3061,6 @@ Dragon4_PrintFloat_IBM_double_double(
     return Format_floatbits(buffer, bufferSize, bigints, exponent1,
                             signbit, mantissaBit, hasUnequalMargins, opt);
 }
-
-#if defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_LE)
-static npy_uint32
-Dragon4_PrintFloat_IBM_double_double_le(
-    Dragon4_Scratch *scratch, npy_float128 *value, Dragon4_Options *opt)
-{
-    FloatVal128 val128;
-    FloatUnion128 buf128;
-
-    buf128.floatingPoint = *value;
-    val128.lo = buf128.integer.a;
-    val128.hi = buf128.integer.b;
-
-    return Dragon4_PrintFloat_IBM_double_double(scratch, val128, opt);
-}
-#endif /* HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_LE */
-
-#if defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_BE)
-static npy_uint32
-Dragon4_PrintFloat_IBM_double_double_be(
-    Dragon4_Scratch *scratch, npy_float128 *value, Dragon4_Options *opt)
-{
-    FloatVal128 val128;
-    FloatUnion128 buf128;
-
-    buf128.floatingPoint = *value;
-    val128.hi = buf128.integer.a;
-    val128.lo = buf128.integer.b;
-
-    return Dragon4_PrintFloat_IBM_double_double(scratch, val128, opt);
-}
-
-#endif /* HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_BE */
 
 #endif /* HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_LE | HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_BE */
 

--- a/numpy/core/src/multiarray/dragon4.h
+++ b/numpy/core/src/multiarray/dragon4.h
@@ -75,10 +75,9 @@
     #define NPY_LONGDOUBLE_BINFMT_NAME Intel_extended128
 #elif defined(HAVE_LDOUBLE_MOTOROLA_EXTENDED_12_BYTES_BE)
     #define NPY_LONGDOUBLE_BINFMT_NAME Motorola_extended96
-#elif defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_LE)
-    #define NPY_LONGDOUBLE_BINFMT_NAME IBM_double_double_le
-#elif defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_BE)
-    #define NPY_LONGDOUBLE_BINFMT_NAME IBM_double_double_be
+#elif (defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_LE) || \
+       defined(HAVE_LDOUBLE_IBM_DOUBLE_DOUBLE_BE))
+    #define NPY_LONGDOUBLE_BINFMT_NAME IBM_double_double
 #else
     #error No long double representation defined
 #endif

